### ISSUE TITLE
Generating States

### ISSUE THREAD

**Issue by onaclov2000** [openai/retro#235]
## How-To
I apologize for making this one an issue, but I wanted to note/mention it here.  I didn't see it obvious in the docs.

I was attempting to integrate Super Metroid into the environment and needed save states, I couldn't find a clear way to get one (though I imagine installing snes9x or something would work), so after searching the code a bit I found the 'get_state' function.

I did a bit of experimenting and discovered I can get state, gzip it and then save it, and it can be loaded just like the states that exist for other games.
````
def save_state_to_file(env, name="test.state"):
    
    content = env.em.get_state()
    with gzip.open(name, 'wb') as f:
        f.write(content)
````

Basically pass the env and a name for the save state, and you can write it out, and load it using the standard "state=<statename>" process when making the game.

**Comment by onaclov2000** [comment]
Closing as this isn't a "valid" issue, aside from perhaps something in docs would be nice. I'm hoping this is easy to find for others who might want to save their state.

**Comment by Jorvan758** [comment]
Excuse me, mate. I followed your method to correctly save the state, but my loading is failing. Does the file .state needs to be in a specific folder and/or does it need other files?

**Comment by onaclov2000** [comment]
Take a look here, I use the above code.
https://github.com/TwitchAIRacingLeague/HackRandomizer/blob/main/play.py

Not sure what you specific issue is, but that seemed to work fine.

**Comment by Jorvan758** [comment]
Thanks a lot. The env.em.set_state() method was what I needed 😉👌
